packages/indexer-common/src/indexer-management/monitor.ts

```diff
@@ -489,6 +490,50 @@ export class NetworkMonitor {
     }
   }
 
+  async transferredDeployments() : Promise<TransferredSubgraphDeployment[]> {
+      try {
+          const result = await this.networkSubgraph.query(
+            gql`
+              query subgraphDeployments {
+                subgraphDeployments(
+                  where: { transferredToL2: true }
+                  orderBy: transferredToL2At
+                  orderDirection: asc
+                ) {
+                  transferredToL2
+                  transferredToL2At
+                  transferredToL2AtTx
+                  transferredToL2AtBlockNumber
+                  ipfsHash
+                }
+              }
+            `,
+          )
+
+        if (result.error) {
+          throw result.error
+        }
+
+        const transferredDeployments = result.data.subgraphDeployments
+
+        // There may be no transferred subgraphs, handle gracefully
+        if (transferredDeployments.length == 0) {
+          this.logger.warn('Failed to query subgraph deployments transferred to L2: no deployments found')
+          throw new Error('No transferred subgraph deployments returned')
+        }
+
+        return transferredDeployments.map((deployment: any) => {
+          deployment.protocolNetwork = this.networkCAIPID
+        })
```

Comment:
I'm afraid this might return an array of undefined values, since the result of the mapped function is an `undefined` value.

Reply:
I think it can be fixed by returning the deployment in the end:
```suggestion
          deployment.protocolNetwork = this.networkCAIPID
          return deployment
```

Reply:
I think it can be fixed by returning the deployment in the end:
```suggestion
        return transferredDeployments.map((deployment: any) => {
          deployment.protocolNetwork = this.networkCAIPID
          return deployment
        })
```

Reply:
yes! good catch, thanks.